### ISSUE TITLE
Complete notification endpoint HTTP client

### ISSUE THREAD

**Issue by affo** [influxdata/influxdb#17023]
See `influxdb/http/notification_endpoint.go`.

The client covers (has methods for soliciting) these URIs:

```
const (
	prefixNotificationEndpoints          = "/api/v2/notificationEndpoints"
	notificationEndpointsIDPath          = "/api/v2/notificationEndpoints/:id"
```

But not these:

```
	notificationEndpointsIDMembersPath   = "/api/v2/notificationEndpoints/:id/members"
	notificationEndpointsIDMembersIDPath = "/api/v2/notificationEndpoints/:id/members/:userID"
	notificationEndpointsIDOwnersPath    = "/api/v2/notificationEndpoints/:id/owners"
	notificationEndpointsIDOwnersIDPath  = "/api/v2/notificationEndpoints/:id/owners/:userID"
	notificationEndpointsIDLabelsPath    = "/api/v2/notificationEndpoints/:id/labels"
	notificationEndpointsIDLabelsIDPath  = "/api/v2/notificationEndpoints/:id/labels/:lid"
)
```

Even if there are handlers mounted for those:

```
h.HandlerFunc("POST", notificationEndpointsIDMembersPath, newPostMemberHandler(memberBackend))
h.HandlerFunc("GET", notificationEndpointsIDMembersPath, newGetMembersHandler(memberBackend))
h.HandlerFunc("DELETE", notificationEndpointsIDMembersIDPath, newDeleteMemberHandler(memberBackend))

...
```

Also it is not clear to me why the HTTP client __is__ also a `UserResourceMappingService` and a `OrganizationService`:

```
type NotificationEndpointService struct {
	Client *httpc.Client
	*UserResourceMappingService
	*OrganizationService
}
```

Those clients executes http requests not to the URIs above, but to the "standard ones"; e.g.

```
func resourceIDPath(resourceType influxdb.ResourceType, resourceID influxdb.ID, p string) string {
	return path.Join("/api/v2/", string(resourceType), resourceID.String(), p)
}
```

Also, those methods cannot be overridden, otherwise, when used in a `*http.Service` they would break the actual implementation of those individual services.

The client should not embed those clients:

```
type NotificationEndpointService struct {
	Client *httpc.Client
	urms *UserResourceMappingService
	os *OrganizationService
}
```

And provide a different prefix URIs to those services. 
Then, provide different methods also to limit the surface of what one can do (URMService and Org actually expose more surface).

**Comment by affo** [comment]
FYI @jsteenb2 I saw you touched that code, do you think my analysis is correct, or am I missing something important?

**Comment by jsteenb2** [comment]
@affo yah, this is one of those things that comes up short in my view. The service doesn't represent the domain per say. It represents the type, which I believe is not the right abstraction. You can extend the http client if you want it to do other things still. I've been in here a couple times to touch things up as I needed for the pkger work. But the reality is the domain isn't encapsulated in the service, it only encapsulates the type :shrug:‍♂ . 

I wrote to this in the service definition proposal. This would be another good example of where the domain can cause some issues. As an aside, if you want to add this as a reference to the service definition, I'm all for it :+1:

**Comment by affo** [comment]
__UPDATE__:

I actually describe why it works in the issue above 😅 .

> Those clients executes http requests not to the URIs above, but to the "standard ones"; e.g.

The "standard ones" are the right ones!

However, I'll update the client to add methods to access those exact urls easily 👍

**Comment by jsteenb2** [comment]
@affo right, the issue becomes the if you're taking this http service to satisfy an interface, those new methods aren't going to do anything 🤷‍♂, they won't be available

**Comment by affo** [comment]
Ice-boxing this because there is not a clear DOD.

Keepin' it open because I still think we need some cleanup in the HTTP client.
